flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -519,6 +519,21 @@ createBoxProcCharTuple(Fortran::lower::AbstractConverter &converter,
                                                      boxProc, charLen);
 }
 
+// Helper to get the ultimate first symbol. This works around the fact that
+// symbol resolution in the front end doesn't always resolve a symbol to its
+// ultimate symbol but may leave placeholder indirections for use and host
+// associations.
```

Comment:
Why is it needed to add `GetUltimate()` calls here if the SymbolMap is now also doing it ? (I suspect there is a good reason, it is just not obvious to me).

Reply:
These helper template functions are extracting the ultimate symbol from an EvExpr. (Just to make the code more concise.)